src/renderer/views/Watch/Watch.js

```diff
@@ -395,8 +396,38 @@ export default Vue.extend({
             if (typeof startTimestamp !== 'undefined') {
               const upcomingTimestamp = new Date(result.videoDetails.liveBroadcastDetails.startTimestamp)
               this.upcomingTimestamp = upcomingTimestamp.toLocaleString()
+
+              let upcomingTimeLeft = upcomingTimestamp - new Date()
+              let timeUnit
+              if (upcomingTimeLeft <= 0) {
+                this.upcomingTimeLeft = this.$t('Video.Starting Soon')
+              } else {
+                upcomingTimeLeft = upcomingTimeLeft / 1000
+                timeUnit = 'Second'
+                if (upcomingTimeLeft > 60) {
+                  upcomingTimeLeft = upcomingTimeLeft / 60
+                  timeUnit = 'Minute'
+                  if (upcomingTimeLeft > 120) {
+                    upcomingTimeLeft = upcomingTimeLeft / 60
+                    timeUnit = 'Hour'
+                    if (upcomingTimeLeft > 24) {
+                      upcomingTimeLeft = upcomingTimeLeft / 24
+                      timeUnit = 'Day'
+                    }
+                  }
+                }
+                upcomingTimeLeft = Math.floor(upcomingTimeLeft)
+                if (upcomingTimeLeft !== 1) {
+                  timeUnit = timeUnit + 's'
+                }
+                timeUnit = this.$t('Video.Published.' + timeUnit)
+
+                // TODO a template might be needed here but idk how they work
+                this.upcomingTimeLeft = upcomingTimeLeft + ' ' + timeUnit.toLowerCase()
```

Comment:
Better temp variable scoping, flatten nested code, rename temp variables, use string intepolation
```suggestion
              if (upcomingTimeLeft <= 0) {
                this.upcomingTimeLeft = this.$t('Video.Starting Soon')
              } else {
                // Convert from ms to second
                upcomingTimeLeft = upcomingTimeLeft / 1000
                let timeUnitI18nPartialKey = 'Second'

                if (upcomingTimeLeft > 60) {
                  upcomingTimeLeft = upcomingTimeLeft / 60
                  timeUnitI18nPartialKey = 'Minute'
                }

	            // Youtube switches to showing time left in minutes at 120 minutes remaining
	            if (timeUnitI18nPartialKey === 'Minute' && upcomingTimeLeft > 120) {
	              upcomingTimeLeft = upcomingTimeLeft / 60
	              timeUnitI18nPartialKey = 'Hour'
	            }
	              
	            if (timeUnitI18nPartialKey === 'Hour' && upcomingTimeLeft > 24) {
	              upcomingTimeLeft = upcomingTimeLeft / 24
	              timeUnitI18nPartialKey = 'Day'
	            }
                
                // Value after decimal not to be displayed
                // e.g. > 2 days = display as `2 days`
                upcomingTimeLeft = Math.floor(upcomingTimeLeft)
                if (upcomingTimeLeft !== 1) {
                  timeUnitI18nPartialKey = timeUnitI18nPartialKey + 's'
                }
                const timeUnitTranslated = this.$t(`Video.Published.${timeUnitI18nPartialKey}`).toLowerCase()

                // TODO a I18n entry for time format might be needed here
                this.upcomingTimeLeft = `${upcomingTimeLeft} ${timeUnitTranslated}`
```

Reply:
Thanks,
I had trouble with naming the time unit variable.

Should string interpolation be used when adding 's' to timeUnitI18nPartialKey, or is that not preferred there?

Reply:
Ya that can be done as well I am just being lazy :P